pkg/agent/openflow/pipeline.go

```diff
@@ -2734,20 +2793,46 @@ func (f *featurePodConnectivity) hostBridgeLocalFlows() []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	return []binding.Flow{
 		// This generates the flow to forward the packets from uplink port to bridge local port.
-		ClassifierTable.ofTable.BuildFlow(priorityNormal).
+		ClassifierTable.ofTable.BuildFlow(priorityLow).
 			Cookie(cookieID).
 			MatchInPort(config.UplinkOFPort).
 			Action().Output(config.BridgeOFPort).
 			Done(),
 		// This generates the flow to forward the packets from bridge local port to uplink port.
-		ClassifierTable.ofTable.BuildFlow(priorityNormal).
+		ClassifierTable.ofTable.BuildFlow(priorityLow).
 			Cookie(cookieID).
 			MatchInPort(config.BridgeOFPort).
 			Action().Output(config.UplinkOFPort).
 			Done(),
 	}
 }
 
+// hostBridgeUplinkVLANFlows generates the flows to match VLAN packets from uplink port.
+func (f *featurePodConnectivity) hostBridgeUplinkVLANFlows() []binding.Flow {
+	vlanMask := uint16(0x1000)
```

Comment:
what does it mean by using 0x1000 mask? Isn't vlan only using 12bits?

Reply:
At here, we use it to match packet with any VLAN, this is required for popVLAN action.
`OXM_OF_VLAN_VID` supports below match cases:

```
              Fully wildcarded
                     Matches any packet, that is, one without an 802.1Q header
                     or with an 802.1Q header with any TCI value.

              Value 0x0000 (OFPVID_NONE), mask 0xffff (or no mask)
                     Matches only packets without an 802.1Q header.

              Value 0x1000, mask 0x1000
                     Matches  any  packet with an 802.1Q header, regardless of
                     VLAN ID.

              Value 0x1009, mask 0xffff (or no mask)
                     Match only packets with an 802.1Q header with VLAN ID 9.

              Value 0x1001, mask 0x1001
                     Matches only packets that have an 802.1Q header  with  an
                     odd-numbered  VLAN  ID. (This is just an example; one can
                     match on any desired VLAN ID bit pattern.)
```

Reply:
Thanks. Fixed.